tests/platformSetup/platformSetup.py

```diff
@@ -344,9 +344,16 @@ def create_tfvars_file(self, test_prefix, image_path=None, cname=None):
             "openstack": "raw",
         }
 
-        # Generate cname if not provided
-        if not cname:
-            cname = self.scripts.get_cname()
+        # Generate image_name if not provided
+        if not image_name:
+            image_name = self.scripts.get_image_name()
+
+        # Determine if we should add extension based on image_path
+        image_source_type = image_path.split("://")[0] if image_path else "file"
```

Comment:
You could use [`urlparse()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) here as well.

Reply:
:+1: using `urlparse()` now